surveys beta docs (#6131)
PostHog/posthog.com: contents/docs/surveys/manual.mdx

---
title: Surveys (beta)
---

> Surveys are a great way to collect qualitative feedback from your users. Qualitative feedback allows you to gain more descriptive insight into your users' experience.

Reviewer: ```suggestion
Surveys are a great way to collect qualitative feedback from your users. Qualitative feedback enables you to gain more insight into your users' and how they view their experience with your product.
```

> Surveys are a great way to collect qualitative feedback from your users. Qualitative feedback enables you to gain more insight into your users' and how they view their experience with your product.

Reviewer 2: ```suggestion
Surveys are a great way to collect qualitative feedback from your users. Qualitative feedback enables you to gain more insight your users' experience with your product.
```

> With surveys you'll be able to ask your users questions, target based on page url, selectors, feature flag, and user properties, and then analyze the results.

Reviewer: ```suggestion
With PostHog you can target surveys based on page url, selectors, feature flag, and user properties. Then, you can analyze the results right in PostHog too!
```

> With PostHog you can target surveys based on page url, selectors, feature flag, and user properties. Then, you can analyze the results right in PostHog too!

Reviewer 2: ```suggestion
With PostHog you can target surveys based on page URL, selectors, feature flag, and user properties. Once filled out, you can analyze the results right in PostHog too!
```

## Getting started

> Setting up surveys is easy! We recommend option 1 if you're just getting started, but if you want to build your own custom survey components, option 2 will let you do that.

Reviewer: ```suggestion
Setting up surveys is easy, and there are two ways to do it. The first option will enable you to get started quickly with a pre-built survey, while the second option involves creating your own UI components.
```

> Setting up surveys is easy, and there are two ways to do it. The first option will enable you to get started quickly with a pre-built survey, while the second option involves creating your own UI components.

Reviewer 2: ```suggestion
Setting up surveys is easy, and there are two ways to do it. The first option enables you to get started quickly with a pre-built survey, while the second option involves creating your own UI components.
```

> ### Option 1 (recommended)

Reviewer 2: ```suggestion
### Option 1: Site app (recommended)
```

> 1. Make sure you've installed and enabled the surveys app in your project's apps.

Reviewer: I'm not fully au fait with how apps are changing at the moment, but we may need to change the language here to make this distinct from apps/integrations/plugins/cdp. 

We should also make these clickable - e.g:

Reviewer: ```suggestion
1. Make sure you've installed and enabled the surveys app in [your project's apps](https://app.posthog.com/apps).
```

Reviewer 2: This won't be a CDP pipeline, it will definitely be an app.

> 2. You must have the `opt_in_site_apps` option enabled in your PostHog's initialization code.

Reviewer: ```suggestion
2. You must have the `opt_in_site_apps` option enabled in your PostHog initialization code.
```

```js
posthog.init("<ph_project_api_key>", { 
    "api_host": "<ph_instance_address>",
    "opt_in_site_apps": true 
})
```

> 3. That's it! Now you can create surveys in the PostHog UI.

Reviewer: This needs to be more explicit. How can I do that? Where?

> ### Option 2 (custom UI component)

Reviewer 2: ```suggestion
### Option 2: Custom UI component
```

> 1. Make sure you're up to date on PostHog's [JavaScript library](/docs/libraries/js), or at least on version 1.67.1.

Reviewer: ```suggestion
1. Make sure you're up to date on PostHog's [JavaScript library](/docs/libraries/js). Surveys requires version 1.67.1, or later.
```

> 1. Make sure you're up to date on PostHog's [JavaScript library](/docs/libraries/js). Surveys requires version 1.67.1, or later.

Reviewer 2: ```suggestion
1. Make sure your installation of PostHog's [JavaScript library](/docs/libraries/js) is update-to-date. Surveys requires version 1.67.1, or later.
```

> 2. You should be able to call `posthog.getSurveys()` and get a list of all surveys back, and `posthog.getActiveMatchingSurveys()` to get a list of active matching surveys that should display for the current user client.

Reviewer: ```suggestion
2. Call `posthog.getSurveys()` and get a list of all surveys back, and `posthog.getActiveMatchingSurveys()` to get a list of active matching surveys that should display for the current user client.
```

```js
posthog.getActiveMatchingSurveys((surveys) => {
    // do something with surveys
})
```

> Note: You'll need to handle whether a user has seen a survey already or not manually if you're using this option. The [survey app](https://github.com/PostHog/feature-surveys/blob/main/site.ts) uses localStorage to keep track of this, which you can implement similarly in your code.

Reviewer 2: ```suggestion
> **Note:** You'll need to handle whether a user has seen a survey already or not manually if you're using this option. The [survey app](https://github.com/PostHog/feature-surveys/blob/main/site.ts) uses `localStorage` to keep track of this, which you can implement similarly in your code.
```

> 3. In order for survey responses to be recorded, you'll need to send in a capture call event with the survey's name and and any properties you'd want.

Reviewer 2: ```suggestion
3. To record survey responses, you need to send in a capture call event with the survey's name and any properties you want.
```
